config/initializers/carrierwave.rb

```diff
@@ -0,0 +1,42 @@
+require 'carrierwave/orm/activerecord'
+
+def setup_for_storing_assets_in_file_system
+  CarrierWave.configure do |config|
+    config.permissions           = 0600
+    config.directory_permissions = 0700
+    config.storage               = :file
+    config.asset_host            = Rails.application.secrets.host
+  end
+end
+
+def setup_for_storing_assets_in_s3
+  CarrierWave.configure do |config|
+    config.root            = Rails.root.join('tmp')
+    config.fog_credentials = {
+        provider:              'AWS',
+        aws_access_key_id:     Rails.application.secrets.aws_s3['access_key_id'],
+        aws_secret_access_key: Rails.application.secrets.aws_s3['secret_access_key'],
+        region:                'us-west-2'
+    }
+    config.fog_directory   = Rails.application.secrets.aws_s3['bucket_name']
+    config.fog_public      = true
+    config.fog_attributes  = { 'Cache-Control' => 'max-age=315576000' }
+    config.storage         = :fog
+    config.cache_dir       = 'carrierwave'
+  end
+end
+
+case Rails.env
+  when 'test'
+    CarrierWave.configure do |config|
+      config.root              = Rails.root.join('tmp')
+      config.storage           = :file
+      config.enable_processing = false
+    end
+  else
+    if Rails.application.secrets.store_uploaded_assets_in == 'aws_s3'
+      setup_for_storing_assets_in_s3
+    else
```

Comment:
Please explictly check for 'filesystem`. If there is a typo in `aws_s3` then it will start using `filesystem`. And add an else clause if there is no match and raise exception to catch typo scenarios.